recipes/libdatrie/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+
+
+class LibdatrieConan(ConanFile):
+    name = "libdatrie"
+    description = "Implementation of double-array structure for representing tries"
+    license = "LGPL-2.1-or-later"
+    homepage = "https://linux.thai.net/projects/datrie"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("trie", "double-array-trie", "datrie")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "tools": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "tools": False,
+    }
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def requirements(self):
+        self.requires("libiconv/1.17")
+
+    def validate(self):
+        if is_apple_os(self) and self.options.shared:
+            # Fails due to build script bugs
+            raise ConanInvalidConfiguration("shared builds on Apple OS-s are not supported. Contributions are welcome!")
+
+    def build_requirements(self):
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
+        if is_msvc(self):
+            self.tool_requires("automake/1.16.5")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.extend([
+            "--disable-doxygen-doc",
+        ])
+        if is_msvc(self):
+            tc.extra_cflags.append("-FS")
+            tc.extra_cxxflags.append("-FS")
+        tc.generate()
+
+        if is_msvc(self):
+            env = Environment()
+            iconv_info = self.dependencies["libiconv"].cpp_info
+            env.append("CPPFLAGS", [f"-I{unix_path(self, iconv_info.includedir)}"] + [f"-D{d}" for d in iconv_info.defines])
+            env.append("_LINK_", [lib if lib.endswith(".lib") else f"{lib}.lib" for lib in iconv_info.libs])
+            env.append("LDFLAGS", [f"-L{unix_path(self, iconv_info.libdir)}"])
+            env.vars(self).save_script("conanautotoolsdeps_cl_workaround")
```

Comment:
Any insight into why this workaround is needed?

Reply:
It's a workaround for the issue described in https://github.com/conan-io/conan/issues/12784